zebra-network/src/peer_set/set.rs

```diff
@@ -724,7 +724,9 @@ where
             // Avoid routing requests to peers that are missing inventory.
             // If we kept trying doomed requests, peers that are missing our requested inventory
             // could take up a large amount of our bandwidth and retry limits.
-            Err(SharedPeerError::from(PeerError::NotFound(vec![hash])))
+            Err(SharedPeerError::from(PeerError::NotFoundRegistry(vec![
```

Comment:
We're generating a local error here, without checking if the peer has new inventory.

We use a special error type, so we only track missing blocks where we actually sent a message to the peer.